### PR TITLE
Misc clean-up Zephyr 3.7 release notes and migration guide

### PR DESCRIPTION
**Pull request by kartben**
- reworked "release highlights" (added PTP, PSA Crypto) and added hyperlinks to relevant sections
- basic restructuredtext/sphinx fixups
- fixed typos / spelling mistakes

This builds on PR #76104 for convenience and to have one less chance of potential merge conflicts. 

If that helps, I can also add to this PR the "final" clean-up consisting in removing sections with empty headers.

### CODE REVIEW

**Comment by kartben** [comment]
rebased to solve merge conflict

**Review by aescolar — APPROVED** [review]
Thanks :) ~rebase needed~

**Comment by aescolar** [comment]
Let's try to get this in relatively fast otherwise it will keep merge conflicting.
(I expect we will need another polishing round closer to Friday due to the remaining release note updates)

File: doc/releases/migration-guide-3.7.rst [hunk 1]
```diff
@@ -263,8 +263,8 @@ Device Drivers and Devicetree
         gyro-odr = <ICM42688_GYRO_ODR_2000>;
     };
 
-* `st,lis2mdl` property `spi-full-duplex` changed to `duplex =
-  SPI_FULL_DUPLEX`. Full duplex is now the default.
+* ``st,lis2mdl`` property ``spi-full-duplex`` changed to ``duplex =
```
Comment (kartben) [inline doc/releases/migration-guide-3.7.rst:266]: Note to self / missed this one

```suggestion
* :dtcompatible:`st,lis2mdl` property ``spi-full-duplex`` changed to ``duplex =
```

File: doc/releases/migration-guide-3.7.rst [hunk 2]
```diff
@@ -591,7 +591,7 @@ Enhanced Serial Peripheral Interface (eSPI)
   ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_STS``, ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_DONE`` and
   ``ESPI_VWIRE_SIGNAL_TARGET_GPIO_<NUMBER>`` respectively to reflect the new terminology
   in eSPI 1.5 specification. (:github:`68492`)
-  The KConfig ``CONFIG_ESPI_SLAVE`` was renamed to ``CONFIG_ESPI_TARGET``, similarly
+  The Kconfig ``CONFIG_ESPI_SLAVE`` was renamed to ``CONFIG_ESPI_TARGET``, similarly
```
Comment (henrikbrixandersen) [inline doc/releases/migration-guide-3.7.rst:594]: Missing :kconfig:option references.

**Review by henrikbrixandersen — APPROVED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -1602,9 +1606,9 @@ Devicetree
 Kconfig
 *******
 
-* Added a `substring` kconfig preprocessor function.
-* Added a `dt_node_ph_prop_path` kconfig preprocessor function.
-* Added a `dt_compat_any_has_prop` kconfig preprocessor function.
+* Added a ``substring`` kconfig preprocessor function.
+* Added a ``dt_node_ph_prop_path`` kconfig preprocessor function.
+* Added a ``dt_compat_any_has_prop`` kconfig preprocessor function.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:1611]: s/kconfig/Kconfig/

**Pull request merged**